imsim/skycat.py

```diff
@@ -52,6 +52,9 @@ def __init__(self, file_name, wcs, bandpass, obj_types=None,
         region = skyCatalogs.Box(*get_radec_limits(wcs, logger, edge_pix)[:4])
         self.objects = sky_cat.get_objects_by_region(region,
                                                      obj_type_set=obj_types)
+        if not self.objects:
+            raise RuntimeError("No skyCatalogs objects on image")
```

Comment:
I don't think this should be an error.  I believe this is the correlate to the place in instcat where you warn about "No objects found on image".  Thus, I think this should also be just a logged warning.

Reply:
OK. But what should I return if the we really hit the invalid access of item 0 in the empty list?
Or how would you prevent reaching this piece of code?

Reply:
What I also thought:
Cannot we interpret `image.nobjects` as an upper limit? Or is there a usecase where a user really wants to have exact that many objects?

Reply:
It's the same thing as you did in instcat, isn't it?  It should only be an error if the user explicitly asks for more objects than exist.  But at this point, it should be allowed for there to be 0 objects on the image.

Reply:
OK. Rebased, added a quick test and turned the above exception into a warning